src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +191,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (!empty($request['name'])) {
+                $options['name'] = $request['name'];
+            }
+
+            if (!empty($request['attached_files'])) {
```

Comment:
@teldosas "`extractFileAttachments` always returns a **non empty** string or null", sorry I missed an important part :)

`isset()` will of course work, but sementically it does not make sense, because the `attached_files` key is always set.
BTW, I tend to be strict about types: `null` means no attached files/no name, a string value means I want a value. 

Then the question is what to do when that value is invalid (e.g. an empty string).

I think we have two options :
- throwing an exception
- let Facebook Graph API returns an error

I personally prefer the last option. It's tedious to handle **all** user's type/value errors, we already pretty well document the usage of our functions :)

ping @SammyK, thoughts?